### DISCUSSION TITLE
Change "user code" to "application" everywhere

### DISCUSSION THREAD

**Discussion by ftessier** [nrc-cnrc/EGSnrc#686] [Ideas]
The term "user code" to describe EGSnrc programs is antiquated and confusing. I suggest we change all instances to refer to "application" or "app" in the code and in the documentation, a concept which is nowadays well understood. There are currently 1560 hard-coded references to "user code" in the repository (about a third of that being in the documentation), which are easy to find:

```
cd $HEN_HOUSE 
grep -iER "user.{1,2}code" *
```

(There are also a few fringe cases with line breaks between "user" and "code".)

There are only 3 file names which have to be updated:

```
./HEN_HOUSE/scripts/compile_user_code
./HEN_HOUSE/scripts/run_user_code
./HEN_HOUSE/scripts/run_user_code_batch
./HEN_HOUSE/user_codes
```

**Comment by crcrewso** [comment]
I agree that user code might be confusing and intimidating but I'm not sure about app or application. 

May I suggest (as a discussion starting point) using <user?> environment for  EGShome  and tools for BEAM, dosxyznrc etc. with scripts for all those tiny helpful applets like exb and clean_after_parallel.

**Comment by ftessier** [comment]
Thanks. The long term outlook is to port EGSnrc entirely ported to C++, at which point every simulation will indeed be an application derived the egs++ class library, for a specific task. So I think "application" is appropriate. As for the little utilities, in time I would like to collect them under a common naming scheme such as

```
egs-clean
egs-run
egs-submit
egs-test
egs-config
egs-plot
...
```

and build a central `egs` disptach tool (not unlike the individual git commands under `git`). All these little tools on the command line would look like subordinate to a top-level `egs` command.

**Comment by rtownson** [comment]
I've heard complaints about the term 'user code' being confusing a number of times. I favor 'application' as being very clear. If I were searching for DOSXYZnrc and saw the folder structure, certainly I would look in 'applications' first. I'm not so sure about 'tools'.

**Comment by randlet** [comment]
I'll add my +1 for the term "application" as well, makes a lot of sense to me.
